packages/edit-site/src/components/sync-state-with-url/use-init-edited-entity-from-url.js

```diff
@@ -54,7 +54,8 @@ function useResolveEditedEntityAndContext( { postId, postType } ) {
 			return {
 				hasLoadedAllDependencies: !! base && !! siteData,
 				homepageId:
-					siteData?.show_on_front === 'page'
+					siteData?.show_on_front === 'page' &&
+					typeof siteData.page_on_front === 'string'
```

Comment:
I don't really understand the fix. if it's a string, there's no need for "toString". toString is necessary for numbers. So I think we should check whether it's null instead.

Reply:
You're right, this slipped through the cracks as I tested things yesterday. The original fix was https://github.com/WordPress/gutenberg/pull/57035/commits/3286416c0cdcf6acf960b48d8cc4099b06f98345, but it would return `undefined` instead of `null`, and while trying to improve it, I ended up wrongly implementing this logic. I'll change it to check against `null`.

Reply:
I wonder if we could simplify it to the following:

```
return {
    hasLoadedAllDependencies: !!base && !!siteData,
    homepageId:
        siteData?.show_on_front === 'page' && typeof siteData?.page_on_front === 'string'
            ? siteData.page_on_front
            : null,
    url: base?.home,
    frontPageTemplateId: _frontPateTemplateId,
};
```

Is there a reason we should be calling `toString` on it? What are the other expected types for this property?

Reply:
yes, there's a reason, unfortunately we have two different APIs, one that give the id as string (from url I guess) and the other as a number. and we have some post ids that are string always (template ids for instance), so we need to cast number to strings to ensure the comparison is correct.

Reply:
I see. So it might be a string or a number? Would the following be okay then?

```js
return {
    hasLoadedAllDependencies: !!base && !!siteData,
    homepageId:
        siteData?.show_on_front === 'page'
            ? (typeof siteData.page_on_front === 'number'
               ? siteData.page_on_front.toString()
               : siteData.page_on_front)
            : null,
    url: base?.home,
    frontPageTemplateId: _frontPateTemplateId,
};
```

Reply:
Yes, I mean I don't like the nested ternaries but yes that works. we could merge strings and numbers because string.toString actually works :P

Reply:
Ah, cool, yeah, good point! I'll push one with the change.

Reply:
@youknowriad done: https://github.com/WordPress/gutenberg/pull/57035/files#diff-90ac1a8edea4ffca3eb4acd2bc2aeed05d54392fc843eb1f253403ae66771dce